text/0000-offline-buildpackages.md

```diff
@@ -0,0 +1,193 @@
+
+This is as simple as specifying any Asset Packages to be used in a build.
+
+`pack build --buildpack some-buildpack -asset asset-package`
+
+The `asset-package` will provide a set of asset files all rooted at `/cnb/assets/` to which buildpacks may use.
+
+As with the `--buildpack` flag, multiple `--asset` flags may be passed. These flags may reference an image, local archive filepath, or URL.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+### Asset Package Definition.
+
+An Asset Package is a collection of layers each of which provide one or more assets. These layers are in lexicographic order by `diffID`. The distribution of assets into layers is an implementation detail left to the platform.
+
+```
+<layer1> ┳━ /cnb/deps/<java11-asset-sha256>/java11
+         ┃                                  ┗ (archive contents)
+         ┗━ /cnb/deps/<java13-asset-sha256>/java13.tgz
+
+...
+<layern> ━━ /cnb/deps/<java15-asset-sha256>/java15.tgz
+```
+
+Asset Packages will have a `io.buildpacks.buildpack.assets` Label. This label's contents will be a `json` object mapping each layer to metadata describing the assets it provides.
+#### Example
+
+``` json
+{
+  "<layer1-diffID>": [
+    {
+      "sha256": "java11-asset-sha256",
+      "uri": "https://path/to/java11.tgz",
+      "additionalPaths": [
+        "other/endpoint1",
+        "other/endpoint2"
+      ],
+      "metadata": {}
+    },
+    {
+      "sha256": "<java13-asset-sha256>",
+      "uri": "/local/path/to/java13.tgz",
+      "additionalPaths": [],
+      "metadata": {
+        "name": "java13"
+      }
+    }
+  ],
+  "...": [],
+  "<layern-diffID>": [
+    {
+      "sha256": "<java15-asset-sha256>",
+      "uri": "/local/path/to/java15.tgz",
+      "additionalPaths": [],
+      "metadata": {}
+    }
+  ]
+}
+```
+
+### Creating Asset Packages.
+
+Asset Image creation will be handled by a new pack command, `pack package-assets --config <packag.toml> <asset-image-name>`.
+
+The `package-assets` command will use a `package.toml` configuration file and produce an `<asset-image-name>` OCI image artifact. `<asset-image-name>` may be a tarball, or a locally available image.
+
+To specify assets included in an Asset Package the `package.toml` file must contain an `assets` array. Each entry in this array may provide:
+  - `uri` (string), (required) local path or URL
+  - `sha256` (string), (required) Must be unique. Used for validation and as an endpoint where an  asset will be provided.
+  - `additional-paths` (array)
+  - `metadata` (arbitrary mapping)
+  - `unzip` (bool), unzip asset when building
+
+The `additional-paths` array lists endpoints where the platform should provide symlinks to the asset. This is provided for cases where `assets` require absolute paths.
+
+The `unzip` field is used to indicate that an asset should be unarchived then added to the Asset Package. In order to use this flag an asset is  must be in one of the following archive formats:
```

Comment:
When I think of `unzip` it makes me think of the `.zip` format explicitly. What do you think of `extract` which is what is used in the [tar manpage](https://linux.die.net/man/1/tar) for GNU flags?